### PR TITLE
Minor JSON config refactor

### PR DESCRIPTION
**Pull request by RealistikDash**
This minor refactor improves the handling of the JSON file by the config module, alongside minorly improving code quality and typing. Solves #27

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -9,119 +15,82 @@
...
-        self.file = new_content
-
+T = TypeVar("T")
 
 class ConfigReader:
     """A parent class meant for the easy management, updating and the creation
     of a configuration `JSON` file."""
+    
+    _json: dict
+    updated_keys: Optional[list[str]] = None
```
Comment (tsunyoku) [inline config.py:25]: being optional is kinda weird, i would just make it non-optional and an empty list and just add to it if keys are updated

File: config.py [hunk 2]
```diff
@@ -9,119 +15,82 @@
...
-    def on_finish_update(self, keys_updated: list):
+    def on_finish_update(self) -> None:
         """Called when the config has just been updated.
         This is meant to be overridden."""
 
         info(
             "The config has just been updated! Please edit according to your preferences!",
         )
-        debug("Keys added: " + ", ".join(keys_updated))
+        debug("Keys added: " + ", ".join(self.updated_keys)) # type: ignore
```
Comment (tsunyoku) [inline config.py:52]: what i said above will fix the need to ignore this type also

File: config.py [hunk 3]
```diff
@@ -9,119 +15,82 @@
...
-        # Check if the key is present. If not, set it.
-        if key not in tuple(self.json.file):
-            # Set it so we can check if the key was modified.
+    
+    def read(self, file_path: str) -> None:
+        """Reads the JSON file and sets it within the object."""
+        
+        if not os.path.exists(file_path):
+            # The updater will handle populating the schema
+            self._json = {}
```
Comment (tsunyoku) [inline config.py:61]: you can initialise this at the top

File: config.py [hunk 4]
```diff
@@ -9,119 +15,82 @@
...
+            dump(config_dict, f, indent=4)
+
+    def get_config_attribute(self, key: str, default: Optional[T]=None) -> Optional[T]:
+        """Fetches the value of the given key from the config, returning the
+        default value."""
+        
+        value = self._json.get(key)
+        
+        if value is None:
+            if not self.updated_keys:
```
Comment (tsunyoku) [inline config.py:87]: what i said at the top will also allow this to go

**Review by tsunyoku — CHANGES_REQUESTED** [review]

**Comment by RealistikDash** [comment]
Upon further inspection, the current implementation of the config is deemed outright incorrect and unsustainable. A complete refactor is thus seen as a preferable alternative.

**Pull request closed without merging**